src/internal/common/sanitize/json_test.go

```diff
@@ -0,0 +1,104 @@
+package sanitize_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/common/sanitize"
+	"github.com/alcionai/corso/src/internal/tester"
+)
+
+type SanitizeJSONUnitSuite struct {
+	tester.Suite
+}
+
+func TestSanitizeJSONUnitSuite(t *testing.T) {
+	suite.Run(t, &SanitizeJSONUnitSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+type jsonTest struct {
+	name        string
+	input       []byte
+	expect      []byte
+	expectValid assert.BoolAssertionFunc
+}
+
+func generateCharacterTests() []jsonTest {
+	var (
+		res []jsonTest
+
+		baseTestName = "Escape0x%02X"
+		baseTestData = `{"foo":"ba%sr"}`
+		expect       = `{"foo":"ba%s\u00%02Xr"}`
```

Comment:
Could you please add some comments around these? e.g. it would help a new reader to know we are expecting escape chars in `\u0000` format here.